src/charm.py

```diff
@@ -140,12 +176,21 @@ def _on_admin_ui_pebble_ready(self, event: WorkloadEvent) -> None:
 
         self._handle_status_update_config(event)
 
+        if not self._container.can_connect():
+            event.defer()
```

Comment:
bit of a random question, in light of the changes to the `defer`  (see [here](https://github.com/canonical/operator/pull/1122)) is this still valid?

Reply:
same applies to all the other `defer` calls

Reply:
That change is currently valid for stop, remove, secret and (pre-)commit events, which are not deferred in admin ui.

On the topic of events deferral, juju 3.4.0 added support for custom pebble notices that we could use instead, but it's a fairly new feature and we first need to move to the later juju and pebble version across our repos.